web-app/js/portal/data/DataCollectionLayerAdapter.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.data');
+
+//
+// This class implements various OpenLayers.Layer functions using a DataCollection (and its associated layers).
+// It allows client code to essentially treat a DataCollection as it would a Layer, without having to use code
+// such as:
+//
+//    dataCollection.getLayerState().getSelectedLayer()...
+//
+// all over the place.
+//
+Portal.data.DataCollectionLayerAdapter = Ext.extend(Ext.util.Observable, {
+
+    constructor: function(config) {
+        Ext.apply(this, config);
+
+        this._layerProperties = {};
+
+        this._copyAttributesFromSelectedLayer();
+        this.layerState.on('selectedlayerchanged', function(newlayer) {
+            console.log('copyAttrs');
+            this._copyAttributesFromSelectedLayer();
+        }, this);
+
+        Portal.data.DataCollectionLayers.superclass.constructor.call(this, config);
+    },
+
+    getSelectedLayer: function() {
+        return this.layerState.getSelectedLayer();
+    },
+
+    _eachLayer: function(fn, scope) {
+        this.layerState._eachLayer(fn, scope);
+    },
+
+    isLoading: function() {
+        return this.getSelectedLayer().loading;
+    },
+
+    _setLayerProperty: function(key, value) {
+        this._getLayerProperties()[key] = value;
+    },
+
+    _getLayerProperty: function(key) {
+        return this._getLayerProperties()[key];
+    },
+
+    _getLayerProperties: function() {
+
+        var layer = this.getSelectedLayer();
+        var key = layer.id;
+
+        if (!this._layerProperties) {
+            this._layerProperties = {};
+        }
+
+        if (this._layerProperties[key] == undefined) {
+            this._layerProperties[key] = {};
+        }
+
+        return this._layerProperties[key];
+    },
+
+    _copyAttributesFromSelectedLayer: function() {
+        Ext.each([
+            'bboxMaxX',
+            'bboxMaxY',
+            'bboxMinX',
+            'bboxMinY',
+            'opacity',
+            'projection'
+        ], function(attr) {
+            this[attr] = this.getSelectedLayer()[attr];
+        }, this);
+    },
+
+    _is130: function() {
```

Comment:
This shouldn't be an underscore method any more, we obviously use it from outside the class (as we've now added it to our adapter). It could probably also use a better name...